### PR TITLE
pyupgrade everything

### PR DESCRIPTION
**Pull request by ajakk**
Signed-off-by: John Helmert III <ajak@gentoo.org>

Anthony Sottile wrote a tool that replaces python2-idiomatic code with python3. This is the result of me running it over the whole of the portage codebase. Most of it is trivial replacements of percent-formatted strings with str().format(), and replacing set and dict instantiations with their literal alternative (ie `set([1,2,3]) -> {1,2,3}`).

### CODE REVIEW

**Comment by Flowdalic** [comment]
Is the tool available somewhere? I am asking because I hoped that it may supports also transformation to f-strings. Or is there a particular reason this did not directly transform to f-strings (which I find more appealing and would prefer to `format()`)?

**Comment by thesamesam** [comment]
> Is the tool available somewhere? I am asking because I hoped that it may supports also transformation to f-strings. Or is there a particular reason this did not directly transform to f-strings (which I find more appealing and would prefer to `format()`)?

It's on pypi, but we can't use f strings as our base version is Python 3.7 for now. We can discuss changing it though.

**Comment by floppym** [comment]
> It's on pypi, but we can't use f strings as our base version is Python 3.7 for now.

f-strings were introduced in Python 3.6.

**Comment by thesamesam** [comment]
> > It's on pypi, but we can't use f strings as our base version is Python 3.7 for now.
> 
> f-strings were introduced in Python 3.6.

Duh. Sorry, I was thinking of f"{...=}".

**Comment by thesamesam** [comment]
dev-python/pyupgrade is now in tree by the way.

File: lib/_emerge/actions.py [hunk 1]
```diff
@@ -1648,7 +1651,8 @@ def create_cleanlist():
 
                 if debug:
                     writemsg_level(
-                        "Candidates: [%s]\n" % ", ".join("'%s'" % (x,) for x in atoms),
+                        "Candidates: [%s]\n"
+                        % ", ".join("'{}'".format(x) for x in atoms),
```
Comment (zmedico) [inline lib/_emerge/actions.py:1655]: Looks like it missed a `.format` opportunity here.

File: lib/_emerge/depgraph.py [hunk 2]
```diff
@@ -2040,7 +2043,9 @@ def _slot_confict_backtrack(self, root, slot_atom, all_parents, conflict_pkgs):
                 "  package(s) to mask: %s" % str(to_be_masked),
                 "      slot: %s" % slot_atom,
                 "   parents: %s"
-                % ", ".join("(%s, '%s')" % (ppkg, atom) for ppkg, atom in all_parents),
+                % ", ".join(
```
Comment (zmedico) [inline lib/_emerge/depgraph.py:2046]: Here's another missed `.format` opportunity.

File: lib/portage/dbapi/bintree.py [hunk 3]
```diff
@@ -575,7 +570,7 @@ def __init__(
             "SLOT": "0",
             "USE": "",
         }
-        self._pkgindex_inherited_keys = ["CHOST", "repository"]
+        self._pkgindex_inherited_keys = ["BINPKG_FORMAT", "CHOST", "repository"]
```
Comment (zmedico) [inline lib/portage/dbapi/bintree.py:573]: This looks like an unintentional revert.

File: lib/portage/update.py [hunk 4]
```diff
@@ -429,11 +425,13 @@ def match_callback(repo_name, atoma, atomb):
                             # add a comment with the update command, so
                             # the user can clearly see what happened
                             contents[pos] = "# %s\n" % " ".join(
```
Comment (zmedico) [inline lib/portage/update.py:427]: Here's another missed `.format` opportunity.

File: lib/_emerge/actions.py (hunk 1 above)
Comment (ajakk) [reply lib/_emerge/actions.py:1655]: Hm, yeah, seems like there's some moderately odd constructs that it doesn't handle:

```
 ~ $ cat pyupgrade-test.py
msg = [
    "%s" % "foo"
]

msg = ["{} {}".format("foo", "%s" % "bar")]
 ~ $ pyupgrade pyupgrade-test.py
 ~ $ cat pyupgrade-test.py
msg = [
    "%s" % "foo"
]

msg = ["{} {}".format("foo", "%s" % "bar")]
```

File: lib/portage/dbapi/bintree.py (hunk 3 above)
Comment (ajakk) [reply lib/portage/dbapi/bintree.py]: Indeed, thanks! Fixed.

File: lib/portage/update.py (hunk 4 above)
Comment (ajakk) [reply lib/portage/update.py]: These 3 should be fixed in a new commit

**Review by zmedico — APPROVED** [review]

File: lib/_emerge/AbstractPollTask.py [hunk 5]
```diff
@@ -93,7 +93,7 @@ def _read_buf(self, fd):
 
     def _async_wait(self):
         self._unregister()
-        super(AbstractPollTask, self)._async_wait()
+        super()._async_wait()
```
Comment (thesamesam) [inline lib/_emerge/AbstractPollTask.py:96]: How did super() work before? (What changed?)

Comment (ajakk) [reply lib/_emerge/AbstractPollTask.py:96]: I am not sure I understand the details, but this pattern is what's recommended in documentation (which says it's the same as the old pattern):

https://docs.python.org/3.11/library/functions.html#super

Comment (thesamesam) [reply lib/_emerge/AbstractPollTask.py:96]: Oh okay, that's all I needed - thanks!

**Comment by thesamesam** [comment]
I think this is good to go but I figure we'll wait until after https://github.com/gentoo/portage/pull/955 is in given the fix for that may require a quick release.

**Pull request closed without merging**